doc/source/data/key-concepts.rst

```diff
@@ -36,75 +37,72 @@ This flexibility is a unique characteristic of Ray Datasets.
 Reading Data
 ============
 
-Datasets uses Ray tasks to read data from remote storage. When reading from a file-based datasource (e.g., S3, GCS), it creates a number of read tasks proportional to the number of CPUs in the cluster. Each read task reads its assigned files and produces an output block:
+Datasets uses Ray tasks to read data from remote storage. Each read task reads its assigned files and produces an output block:
 
 .. image:: images/dataset-read.svg
    :align: center
 
 ..
   https://docs.google.com/drawings/d/15B4TB8b5xN15Q9S8-s0MjW6iIvo_PrH7JtV1fL123pU/edit
 
-The parallelism can also be manually specified, but the final parallelism for a read is always capped by the number of files in the underlying dataset. See the :ref:`Creating Datasets Guide <creating_datasets>` for an in-depth guide
-on creating datasets.
+You can manually specify the number of read tasks, but the final parallelism is always capped by the number of files in the underlying dataset.
```

Comment:
Nit: Should we still mention that the default parallelism will be proportional to the number of CPUs in the cluster? I feel like this is useful to convey, but not sure if it should be in the key concepts or just the creating datasets feature guide.

Reply:
The default parallelism isn't always proportional to the number of CPUs, right? For example, if there are fewer files than CPUs, then the parallelism is the number of files. 

Related discussion: https://github.com/ray-project/ray/pull/33066#discussion_r1127102631